pkg/frontend/admin_hive_clusterdeployment_get.go

```diff
@@ -0,0 +1,69 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/ugorji/go/codec"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) getAdminHiveClusterDeployment(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
```

Comment:
should we check if the typecast is successful?

Reply:
I don't see the harm in checking, however most if not all the other api handlers do this type assertion without checking if it was ok. The check is kind of clunky, do you think we need to add it?

```golang
log, ok := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
if !ok {
	api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal Server Error")
	return
}

Reply:
We can decide for a different approach later on for all handlers. Lets keep this one concise.